subsys/net/lib/aws_fota/src/aws_fota.c

```diff
@@ -434,28 +449,51 @@ int aws_fota_mqtt_evt_handler(struct mqtt_client *const client,
 	return 0;
 }
 
-static void http_fota_handler(enum fota_download_evt_id evt)
+static void wait_for_update_accepted(void)
+{
+	accepted = false;
+	while (!accepted) {
+		k_sleep(K_NO_WAIT);
+	};
+}
+
+static void http_fota_handler(const struct fota_download_evt *evt)
 {
 	__ASSERT_NO_MSG(c != NULL);
 
 	int err = 0;
 
-	switch (evt) {
+	switch (evt->id) {
 	case FOTA_DOWNLOAD_EVT_FINISHED:
 		LOG_INF("FOTA download completed evt received");
 		fota_state = APPLY_UPDATE;
 		err = update_job_execution(c, job_id, AWS_JOBS_IN_PROGRESS,
-				     fota_state, execution_version_number, "");
+					   fota_state,
+					   stored_offset,
+					   execution_version_number, "");
+		wait_for_update_accepted();
```

Comment:
Could these calls be removed, and replaced by a single call at the start of `update_job_execution`?

Reply:
Maybe I'll try and see how it goes. There are some calls which does not need this but it could be that it doesn't hurt the overall program flow.

Reply:
No, didn't seem to work. It seems that if you do this when it's called inside it's own thread it hangs. So it's only used in callbacks received from the download_client thread(`fota_download`).

Reply:
One idea could be to add a wait_for_update_accepted flag to the API call of `update_job_execution` not sure what is better.

Reply:
but it's a good idea to see if it can be extracted out. I've tried different things and haven't gotten it working.

Reply:
Extracted into `update_job_execution`